front: harmonize train naming conventions in tests

Maymanaf opened the pull request:
I tried to update everything I could see. I may have missed some names, so feel free to request any changes

Closes #12185

Waiting for #12399

Maymanaf committed:
front: harmonize train naming conventions in tests

Signed-off-by: maymanaf <med.aymen.naf@gmail.com>

SharglutDev commented on front/src/modules/timetableItem/components/Timetable/TimetableToolbar.tsx:
```diff
@@ -249,7 +249,7 @@ const TimetableToolbar = ({
           })}
         >
-          <div className="train-count" data-testid="timetable-train-count">
+          <div className="train-count" data-testid="timetable-items-count">
```
nit : in english, `count` already implies that there will be multiple items
```suggestion
          <div className="train-count" data-testid="timetable-item-count">
```

SharglutDev commented on front/tests/008-op-times-and-stops-tab.spec.ts:
nit : we could update the comment l.94 to say `Setup train schedule configuration`

SharglutDev commented on front/tests/009-op-simulation-settings-tab.spec.ts:
Same as previous comment for l.139 here

SharglutDev commented on front/tests/pages/operational-studies/scenario-timetable-section.ts:
```diff
@@ -337,44 +334,39 @@ class ScenarioTimetableSection extends OpSimulationResultPage {
+    await this.projectItemButton.nth(index).click();
   }
 
   async getTrainArrivalTime(expectedArrivalTime: string, index = 0) {
```
```suggestion
  async getTimetableItemArrivalTime(expectedArrivalTime: string, index = 0) {
```

SharglutDev commented on front/tests/pages/operational-studies/operational-studies-page.ts:
```diff
@@ -144,9 +138,10 @@ class OperationalStudiesPage extends CommonPage {
 
-  async validateAndCloseTrainEdition() {
-    await this.editTrainButton.click();
+  async submitTrainEdit() {
```
```suggestion
  async submitTimetableItemEdit() {
```

SharglutDev commented on front/tests/pages/operational-studies/scenario-page.ts:
```diff
@@ -155,7 +155,7 @@ class ScenarioPage extends CommonPage {
 
-  async toggletrainList(isOpen: boolean = true) {
+  async toggleTrainList(isOpen: boolean = true) {
     await this.trainsButton.click();
```
Can't seem to find the `data-testid` for this button ? 🤔

Maymanaf replied:
` data-testid={`${board}-button`}` for the class `className="board-btns"`

SharglutDev commented on front/tests/015-train-timetable.spec.ts:
```diff
@@ -88,33 +88,33 @@ test.describe('Verify train schedule elements and filters', () => {
   /** *************** Test 1 **************** */
   test('Loading trains and verifying simulation result for trainSchedule', async () => {
```
Same for l.101 (guess we could change also some `train` in l.102 as well)
```suggestion
  test('Loading timetable items and verifying simulation result for trainSchedule', async () => {
```

SharglutDev commented on front/tests/015-train-timetable.spec.ts:
```diff
@@ -192,25 +192,28 @@ test.describe('Verify train schedule elements and filters', () => {
+    );
+    await scenarioTimetableSection.filterTrainTypeAndVerifyTrainCount('All', TOTAL_TIMETABLE_ITEMS);
 
     // Verify train composition filters with predefined filter codes and expected counts
```
Same on l.210
```suggestion
    // Verify timetable item composition filters with predefined filter codes and expected counts
```

SharglutDev commented on front/tests/015-train-timetable.spec.ts:
```diff
@@ -12,24 +12,24 @@ import {
+  VALID_ITEMS as VALID_TIMETABLE_ITEMS,
   VALID_PACED_TRAINS,
-  VALID_TRAINS,
+  VALID_TRAINS as VALID_TRAIN_SCHEDULE,
```
Don't we want to rename them in the `const` file ?

Maymanaf replied:
Yes, my bad  I did make the change in the `const` file, but I guess I lost it when I rebased

SharglutDev commented on front/tests/015-train-timetable.spec.ts:
```diff
@@ -12,24 +12,24 @@ import {
   TOTAL_PACED_TRAINS,
-  TOTAL_TRAINS,
-  VALID_ITEMS,
+  TOTAL_TRAINS as TOTAL_TRAIN_SCHEDULE,
```
```suggestion
  TOTAL_TRAINS as TOTAL_TRAIN_SCHEDULES,
```

SharglutDev commented on front/tests/017-paced-train-management.spec.ts:
We could rename `train` in l.217

SharglutDev commented on front/tests/018-train-edition.spec.ts:
l.52 `trains` > `train schedules`

SharglutDev commented on front/tests/019-scenario-page-synchronization.spec.ts:
```diff
@@ -80,10 +79,6 @@ test.describe('Synchronize the scenario page across multiple windows', () => {
     }
   );
 
-  test.afterAll('Delete the created scenario', async () => {
-    await deleteScenario(project.id, study.id, scenarioItems.name);
-  });
-
```
This should be in a separate commit if it's not need anymore

SharglutDev commented on front/tests/pages/operational-studies/operational-studies-page.ts:
```diff
@@ -78,21 +76,17 @@ class OperationalStudiesPage extends CommonPage {
+    this.timetableItemTagsInput = page.getByTestId('chips-input');
   }
 
   // Click on the button to add a scenario train.
```
```suggestion
  // Click on the button to add a scenario timetable item.
```

SharglutDev commented on front/tests/pages/operational-studies/scenario-timetable-section.ts:
In this file, we could rename :  
- locator `trainArrivalTime` and `trainArrivalTimeLoader` (and their `data-testid`)
- l.113 `train` > `train schedule` and l.114 `trainSelector` > `trainScheduleSelector`

SharglutDev commented on front/tests/pages/operational-studies/scenario-timetable-section.ts:
```diff
@@ -18,11 +18,9 @@ const frTranslations: ScenarioTranslations = readJsonFile<{
 
 class ScenarioTimetableSection extends OpSimulationResultPage {
-  private readonly invalidTrainsMessage: Locator;
+  private readonly invalidTimetableItemsMessage: Locator;
```
I don't mind renaming this one but we kinda open the pandora box. 
We could then rename the `data-testid`, classNames, ids in filters...

Maymanaf replied:
You're right, but there are more `classNames`, `ids` etc., in the code that require similar changes.
For this case, I’ll update the `data-testid`, but the others should be handled in separate PRs

theocrsb commented on front/src/modules/timetableItem/components/Timetable/TrainScheduleItem.tsx:
here I think we can keep train or trainSchedule

Maymanaf replied:
I think the correct usage should be `timetable-item`, but it’s true that the files aren’t harmonized in terms of `classNames`,` ids`, etc. And as @SharglutDev  mentioned:

> we kinda open the Pandora’s box

If you think it’s better to keep things as they are, I can revert the changes.

Maymanaf committed:
front: remove unnecessary afterAll cleanup

The scenario is already deleted during the test execution, so the afterAll block is redundant.

Signed-off-by: maymanaf <med.aymen.naf@gmail.com>

SharglutDev reviewed (approved):
Lgtm ✅ Thanks for this painful task !

Yohh commented on front/tests/assets/constants/timetable-items-count.ts:
```diff
@@ -1,36 +1,36 @@
 // Constants for expected train counts
```
nit:
```suggestion
```

Yohh commented on front/tests/pages/operational-studies/scenario-timetable-section.ts:
```diff
@@ -106,16 +103,16 @@ class ScenarioTimetableSection extends OpSimulationResultPage {
-  // Get the button locator of a train element.
-  private static getTrainButton(trainSelector: Locator): Locator {
-    return trainSelector.getByTestId('scenario-timetable-train-button');
+  // Get the button locator of a train schedule element.
```
nit: this comment and several others seems useless, the function name and signature talk for herself, not really an issue, you can avoid this comment if you want

Maymanaf committed:
front: remove unnecessary comments

Signed-off-by: maymanaf <med.aymen.naf@gmail.com>
